src/cljs/main/broadfcui/common/sync.cljs

```diff
@@ -0,0 +1,103 @@
+(ns broadfcui.common.sync
+  (:require
+   [dmohs.react :as react]
+   [clojure.string :as string]
+   [broadfcui.common.components :as comps]
+   [broadfcui.common.style :as style]
+   [broadfcui.components.modals :as modals]
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.utils :as utils]
+   ))
+
+
+(defn get-method-display [{:keys [namespace name snapshotId]}]
+  (str namespace "/" name " snapshot " snapshotId))
+
+
+(react/defc SynchronizeModal
+  {:render
+   (fn [{:keys [props state this]}]
+     [modals/OKCancelForm
+      {:header "Synchronize Access to Method"
+       :dismiss (:dismiss props)
+       :content
+       (react/create-element
+        [:div {:style {:maxWidth 670}}
+         [:div {:style {:marginBottom "1.5rem"}}
+          "In order to allow other users of this workspace to run this method, you will need
+           to grant them access to "
+          [:b {} (get-method-display (:method props))]
+          "."]
+         (if (:customize? @state)
+           (this :-render-permission-detail)
+           (style/create-link {:text "Customize Permissions"
+                               :onClick #(this :-setup-permission-detail)}))
+         [comps/ErrorViewer {:error (:grant-error @state)}]])
+       :ok-button [comps/Button {:text (if (:customize? @state)
+                                         "Grant Permission"
+                                         "Grant Read Permission")
+                                 :onClick #(this :-grant-permission)}]}])
+   :-render-permission-detail
+   (fn [{:keys [props locals]}]
+     [:div {}
+      [:p {} "Users currently without access:"]
+      [:table {}
+       [:tbody {}
+        (map (fn [user]
+               [:tr {}
+                [:td {:style {:verticalAlign "baseline"}} user]
+                [:td {:style {:verticalAlign "baseline"}}
+                 [:div {:style {:marginLeft "1rem"}}
+                  (style/create-identity-select
+                   {:style {:width 120}
+                    :onChange #(swap! locals assoc user (.. % -target -value))}
+                   ["Reader" "Owner" "No access"])]]])
```

Comment:
Is `"No access"` in this list because a user might want to grant to only a subset of the suggested people?